src/main/java/com/redhat/devtools/lsp4ij/features/workspaceSymbol/LSPWorkspaceImplementationsSearch.java

```diff
@@ -0,0 +1,142 @@
+
+/**
+ * Implements the IDE's standard Go To Implementation(s) action using LSP textDocument/implementation. -->
+ */
+public class LSPWorkspaceImplementationsSearch implements QueryExecutor<PsiElement, DefinitionsScopedSearch.SearchParameters> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LSPWorkspaceImplementationsSearch.class);
+
+    @Override
+    public boolean execute(@NotNull DefinitionsScopedSearch.SearchParameters queryParameters, @NotNull Processor<? super PsiElement> consumer) {
+        Project project = queryParameters.getProject();
+        if (project.isDisposed()) {
+            return true;
+        }
+
+        PsiElement element = queryParameters.getElement();
+        if (!element.isValid()) {
+            return true;
+        }
+
+        PsiFile file = element.getContainingFile();
+        if ((file == null) || !file.isValid()) {
+            return true;
+        }
+
+        if (ProjectIndexingManager.canExecuteLSPFeature(file) != ExecuteLSPFeatureStatus.NOW) {
+            // The file is not associated to a language server
+            return true;
+        }
+
+        // Check if the file can support the feature
+        if (!LanguageServiceAccessor.getInstance(project)
+                .hasAny(file.getVirtualFile(), ls -> ls.getClientFeatures().getImplementationFeature().isImplementationSupported(file))) {
+            return true;
+        }
+
+        Document document = PsiDocumentManager.getInstance(project).getDocument(file);
+        if (document == null) {
+            return true;
+        }
+
+        int offset = element.getTextRange().getStartOffset();
+        LSPImplementationParams params = new LSPImplementationParams(
+                LSPIJUtils.toTextDocumentIdentifier(file.getVirtualFile()),
+                LSPIJUtils.toPosition(offset, document),
+                offset
+        );
+        LSPImplementationSupport implementationSupport = LSPFileSupport.getSupport(file).getImplementationSupport();
+        CompletableFuture<List<Location>> implementationsFuture = implementationSupport.getImplementations(params);
+        try {
+            waitUntilDone(implementationsFuture, file);
+        } catch (ProcessCanceledException ex) {
+            // cancel the LSP requests textDocument/implementation
+            implementationSupport.cancel();
+        } catch (CancellationException ex) {
+            // cancel the LSP requests textDocument/implementation
+            implementationSupport.cancel();
+        } catch (ExecutionException e) {
+            LOGGER.error("Error while consuming LSP 'textDocument/implementation' request", e);
+        }
+
+        if (isDoneNormally(implementationsFuture)) {
+            List<Location> implementations = implementationsFuture.getNow(null);
+            if (ContainerUtil.isEmpty(implementations)) {
+                // No implementations found
+                Editor editor = PsiEditorUtil.findEditor(file);
```

Comment:
To be consistent with LSP4IJ base code, write:

`Editor editor = LSPIJUtils.editorForElement(file);`